runtime/lua/vim/lsp/buf.lua

```diff
@@ -813,6 +814,16 @@ function M.code_action(options)
       params = util.make_range_params(win, client.offset_encoding)
     end
     params.context = context
+
+    -- Don't send markup to clients that don't support it.
```

Comment:
@mfussenegger sounds like we instead want to filter these diagnostics entirely from servers that don't support markup (instead of changing the message)? I'm not entirely sure of what we agreed in the protocol discussion.

Reply:
Yes they need to be filtered. But the specification should also get an update with this additional clarification.

Reply:
Sounds good. I'm adding the clarification [here](https://github.com/microsoft/language-server-protocol/pull/1917).